### DISCUSSION TITLE
How to checkout and build a previous commit of mrs

### DISCUSSION THREAD

**Discussion by MartinSegaert** [ctu-mrs/mrs_uav_system#160] [Q&A]
Hello,

I recently installed the latest version of ctu-mrs on a new nuc. Then I was able to build the mrs_workspace properly. However, I wanted to go back to a previous commit in "ctu-mrs/simulation" (14 February: c7f63a0c3a4acdcded308554303f0985443ad3ee). This led to an error when building after having checkout to the desired commit:

"""
Errors     << px4:make /home/nuc5/mrs_workspace/logs/px4/build.make.004.log                                                                                                                               
make[2]: *** [src/lib/version/CMakeFiles/ver_gen.dir/build.make:75: src/lib/version/build_git_version.h] Error 1
make[1]: *** [CMakeFiles/Makefile2:8099: src/lib/version/CMakeFiles/ver_gen.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:166: all] Error 2
cd /home/nuc5/mrs_workspace/build/px4; catkin build --get-env px4 | catkin env -si  /usr/bin/make --jobserver-auth=3,4; cd -
"""

![image](https://user-images.githubusercontent.com/62548016/232434625-752695ca-b790-4274-b0f5-859d67f30243.png)

I also tried to ./install.sh inside the installation folder of ctu-mrs/simulation. It gave the same error.

My question is: How am I supposed to go to another commit than master in a one of your repository and then be able to build. Should I do something other than simply checkout and catkin build ? Additionally, as we have both ctu-mrs/simulation and ctu-mrs/uav_core in the mrs_workspace, should these to be compatible commits or can one of them stay on master and the other one on any other previous commit ?

Thanks for your response,

Martin

**Comment by klaxalk** [comment]
Hey, sorry for the large delay... a lot of stuff happening now...

Simply checking out the desired version in the meta-repository + `gitman install` should be typically enough.

**However:** The simulation repository contains the px4 firmware, which does not handle switches between major versions without a clean build (sometimes, even more, is needed https://github.com/ctu-mrs/mrs_uav_system/issues/150). Somewhere near April, we bumped px4 to the new version, which means the clean build needs to be done in either direction: upgrading or downgrading.
